appcues/src/main/java/com/appcues/data/mapper/trait/TraitsMapper.kt

```diff
@@ -14,7 +15,16 @@ internal class TraitsMapper(
         return arrayListOf<ExperienceTrait>().apply {
             from.forEach {
                 traitRegistry[it.first.type]?.also { factory ->
-                    add(factory.invoke(it.first.config, it.second, renderContext))
+                    try {
+                        add(factory.invoke(it.first.config, it.second, renderContext))
+                    } catch (ex: InstanceCreationException) {
+                        // since Traits are loaded through Koin, we catch any issue
+                        // with Koin instance creation here, then hopefully get the underlying
+                        // AppcuesTraitException as the cause and throw that instead. This is
+                        // because higher level mapping code handles the trait exceptions and
+                        // reports proper ExperienceErrors
+                        throw ex.cause ?: ex
```

Comment:
explained in the comments, but this is what supports ModalTrait (or any other) throwing an exception if it's config is invalid